homeassistant/components/rainmachine/__init__.py

```diff
@@ -364,54 +271,20 @@
 class RainMachine:
     """Define a generic RainMachine object."""
 
-    def __init__(
-        self, client, binary_sensor_conditions, sensor_conditions, default_zone_runtime
-    ):
+    def __init__(self, client, default_zone_runtime):
         """Initialize."""
-        self.binary_sensor_conditions = binary_sensor_conditions
         self.client = client
         self.data = {}
         self.default_zone_runtime = default_zone_runtime
         self.device_mac = self.client.mac
-        self.sensor_conditions = sensor_conditions
 
     async def async_update(self):
         """Update sensor/binary sensor data."""
-
-        tasks = {}
-
-        if TYPE_FLOW_SENSOR in self.binary_sensor_conditions or any(
```

Comment:
So now you're fetching all data. One more improvement that you can do in a future PR is that each entity will let the data class know what data it is interested in. That way if all entities of a certain type are disabled, you will no longer fetch the data.

Reply:
This is how I've done it for Ring: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/ring/__init__.py#L173-L235

Reply:
How do you listen for the entity being enabled disabled?

Reply:
`async_added_to_hass` and `async_will_remove_from_hass`?

Reply:
Yeah, use the lifecycle methods to opt-in for data. If data is not being fetched yet, that's when you start fetching that data. In your case you could just track sensor types and count how many need them. Ring has a variant on that too: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/ring/__init__.py#L238